api/src/handlers.rs

```diff
@@ -273,6 +274,46 @@ impl Handler for ChainHandler {
 	}
 }
 
+// Gets block details given either a hex address or height.
+// GET /v1/block/<address>
+// GET /v1/block/<height>
+pub struct BlockHandler {
+	pub chain: Arc<chain::Chain>,
+}
+
+impl BlockHandler {
+	fn get_block(&self, h: &Hash) -> BlockPrintable {
+		let block = self.chain.clone().get_block(h).unwrap();
+		BlockPrintable::from_block(&block)
+	}
+
+	// Try to decode the string as a height or a hash address.
+	fn parse_input(&self, input: String) -> Hash {
+		if let Ok(height) = input.parse() {
+			return self.chain
+				.clone()
+				.get_header_by_height(height)
```

Comment:
This should address the `NotFoundErr` case so we return a nice HTTP error code `404`.